File: docs/guides/examples/cpp/src/data_replication_create.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

int main() {
    // Create a client with the server URL
    auto client = IClient::Build("http://127.0.0.1:8383", {
            .api_token = "my-token"
    });

    // Create a bucket with the name "my-bucket" and a FIFO quota of 1GB
    IBucket::Settings settings;
    settings.quota_type = IBucket::QuotaType::kFifo;
    settings.quota_size = 1'000'000'000;
    auto [bucket, create_err] = client->GetOrCreateBucket("my-bucket", settings);
    assert(create_err == Error::kOk);

    // Set up a replication to a destination bucket for records from the "py-example" entry and with labels "anomaly=1"
    auto repl_err = client->CreateReplication("my-replication", IClient::ReplicationSettings{
        .src_bucket = "my-bucket",
        .dst_bucket = "demo",
        .dst_host = "https://play.reduct.store",
        .dst_token = "reductstore",
        .entries = {"cpp-example"},
        .include = {{"anomaly", "1"}},
    });
    assert(repl_err == Error::kOk);
}

File: docs/examples/go/src/data_replication_create.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	model "github.com/reductstore/reduct-go/model"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})
	// Create a bucket as source bucket
	_, err := client.CreateOrGetBucket(context.Background(), "my-bucket", nil)
	if err != nil {
		panic(err)
	}

	// Set up a replication to a destination bucket for records
	// from the "go-example" entry and with labels "anomaly=1"
	replicationSettings := model.ReplicationSettings{
		SrcBucket: "my-bucket",
		DstBucket: "demo",
		DstHost:   "https://play.reduct.store",
		DstToken:  "reductstore",
		Entries:   []string{"go-example"},
		When: map[string]any{
			"&anomaly": map[string]any{
				"$eq": 1,
			},
		},
	}

	err = client.CreateReplicationTask(context.Background(), "my-replication", replicationSettings)
	if err != nil {
		panic(err)
	}
}
